index.bs

```diff
@@ -644,31 +823,67 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 		<h4 method for="Clipboard">readText()</h4>
 		The {{Clipboard/readText()}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
-				1. Let |r| be the result of running [=check clipboard read permission=] [=in parallel=]
+				1. Let |r| be the result of running [=check clipboard read permission=].
 
-				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
+				1. If |r| is false, then:
+				
+					1. [=Queue a global task=] on the [=permission task source=], given |realm|'s [=Realm/global object=], to [=reject=] |p| with {{"NotAllowedError"}} {{DOMException}} in |realm|.
 
-				1. Let |data| be a copy of the [=system clipboard data=].
+					1. Abort these steps.
 
-				1. Let |textData| be an empty string.
+				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the sanitized copy of text/plain format.
```

Comment:
The type here doesn't match the type of `dataList` below. Either both should be `ClipboardItem`s or `clipboard item`s. It's simpler to change `dataList` to a sequence of `clipboard item`s.

Reply:
Ah, nice catch. Thanks!